guides/schema/class_based_api.md

```diff
@@ -0,0 +1,353 @@
+
+Object fields can be created with the `field(...)` class method, which accepts the similar arguments as the previous `field(...)` method.
+
+```ruby
+# first, somewhere, a base class:
+class Types::BaseObject < GraphQL::Schema::Object
+end
+
+# then...
+class Types::TodoList < BaseObject
+  field :name, String, "The unique name of this list", null: false
+  # Related Object:
+  field :owner, Types::User, "The creator of this list", null: false
+  # List field:
+  field
+  # Connection:
+  field :items, Types::TodoItem.connection_type, "Tasks on this list", null: false do
+    argument :status, TodoStatus, "Restrict items to this status", null: true
+  end
+end
+```
+
+### New return type & argument type specification
+
+The second argument to `field(...)` is the return type. This can be:
+
+- A GraphQL type object built with `.define { ... }`
+- A GraphQL type class which you defined
+- A Ruby built-in such as `Integer`, `Float` or `String`
+- A string which corresponds to a GraphQL built-in, such as `"ID"`, `"Boolean"`, or `"Int"`
+- An _array_ of any of the above, which denotes a list type. Inner list types are always made non-null.
+
+Nullability is expressed with the required `null:` keyword:
+
+- `null: true` means that the field _may_ return null
+- `null: false` means the field is non-nullable; it may not return null. If the implementation returns `nil`, GraphQL-Ruby will return an error to the client.
+
+Here are some examples:
+
+```ruby
+field :name, String, null: true # String
+field :id, "ID", null: false # ID!
+field :scores, [Integer], null: false, # [Int]!
+field :teammates, [Types::User], null: false, # [User]!
+```
+
+### Connection types
+
+There is no `connection(...)` method. Instead, connection fields are inferred from the type name.
+
+If the type name ends in `Connection`, the field is treated as a connection field.
+
+This default may be overridden by passing a `connection: true` or `connection: false` keyword.
+
+### Resolve function compatibility
+
+If you define a type with a class, you can use existing GraphQL-Ruby resolve functions with that class, for example:
+
+```ruby
+# Using a Proc literal or #call-able
+field :something, ... resolve: ->(obj, args, ctx) { ... }
+# Using a predefined field
+field :doSomething, field: Mutations::DoSomething.field
+# Using a GraphQL::Function
+field :something, function: Functions::Something.new
+```
+
+When using these resolution implementations, they will be called with the same `(obj, args, ctx)` parameters as before.
+
+### Resolution with methods
+
+If you implement a field by defining a method, you should expect some automatic transformations:
+
+- GraphQL arguments will be converted to Ruby keyword arguments.
+- If the field name is `camelCased`, the method name should be `underscore_cased`.
+- If any argument names are `camelCased`, they will be passed to the method as `underscore_cased` Ruby keyword args.
```

Comment:
🤘

Reply:
Sorry if I'm jumping the gun here since you just committed these docs, but is the intent to also perform this auto underscore_casing for all input objects (recursively?)

Reply:
yes, the intent is to do it recursively! I'll definitely touch on that more in the "Input Objects" section in this doc.